src/www/ui/admin-obligation-file.php

```diff
@@ -0,0 +1,443 @@
+      $ob .= "<td align=center><a href='";
+      $ob .= Traceback_uri();
+      $ob .= "?mod=" . $this->Name .
+           "&ob_pk=$row[ob_pk]' >'".
+           "<img border=0 src='" . Traceback_uri() . "images/button_edit.png'></a></td>";
+
+      $ob .= "<td align=left>$row[ob_topic]</td>";
+      $vetext = htmlspecialchars($row['ob_text']);
+      $ob .= "<td><textarea readonly=readonly rows=3 cols=40>$vetext</textarea></td> ";
+      $ob .= "<td align=center>$associatedLicenses</td>";
+      $ob .= "</tr>";
+    }
+    pg_free_result($result);
+    $ob .= "</table>";
+    return $ob;
+  }
+
+  /**
+   * @brief Update forms
+   * @param int $ob_pk - for the obligation to update, empty to add
+   * @return string The input form
+   */
+  function Updatefm($ob_pk)
+  {
+    $vars = array();
+
+    $ob_pk_update = "";
+
+    if (0 < count($_POST)) {
+      $ob_pk_update = $_POST['ob_pk'];
+      if (!empty($ob_pk)) $ob_pk_update = $ob_pk;
+      else if (empty($ob_pk_update)) $ob_pk_update = $_GET['ob_pk'];
+    }
+
+    $vars['actionUri'] = "?mod=" . $this->Name."&ob_pk=$ob_pk_update";
+
+    if ($ob_pk)  // true if this is an update
+    {
+      $row = $this->dbManager->getSingleRow("SELECT * FROM ONLY obligation_ref WHERE ob_pk=$1", array($ob_pk),__METHOD__.'.forUpdate');
+      if ($row === false)
+      {
+        $text = _("No obligation matching this key");
+        $text1 = _("was found");
+        return "$text ($ob_pk) $text1.";
+      }
+
+      $associatedLicenses = $this->obligationMap->getLicenseList($ob_pk);
+      $vars['licnames'] = explode(";",$associatedLicenses);
+    }
+    else
+    {
+      $row = array('ob_active' =>'t', 'ob_text_updatable'=>'t');
+    }
+
+    foreach(array_keys($row) as $key)
+    {
+      if (array_key_exists($key, $_POST))
+      {
+        $row[$key] = $_POST[$key];
+      }
+    }
+
+    $vars['boolYesNoMap'] = array("true"=>"Yes", "false"=>"No");
+    $row['ob_active'] = $this->dbManager->booleanFromDb($row['ob_active'])?'true':'false';
+    $row['ob_text_updatable'] = $this->dbManager->booleanFromDb($row['ob_text_updatable'])?'true':'false';
+    $vars['isReadOnly'] = !(empty($ob_pk) || $row['ob_text_updatable']=='true');
+
+    $vars['obId'] = $ob_pk?:$ob_pk_update;
+
+    // get list of known license shortnames
+    $licenseMap = new LicenseMap($this->dbManager, 0, LicenseMap::REPORT);
+    $reportLicenses = $licenseMap->getTopLevelLicenseRefs();
+    $vars['licenseShortnames'] = array();
+    foreach ($reportLicenses as $licRef)
+    {
+      $vars['licenseShortnames'][$licRef->getShortName()] = $licRef->getShortName();
+    }
+    natcasesort($vars['licenseShortnames']);
+
+    $vars['licenseSelectorName'] = 'licenseSelector[]';
+    $vars['licenseSelectorId'] = 'licenseSelectorId';
+    $scripts = "<script src='scripts/tools.js' type='text/javascript'></script>
+      <script src='scripts/select2.full.min.js'></script>
```

Comment:
@max-wittig This is where the select2 function is integrated into the obligation template